### PR TITLE
catch setting of coordiantes and bounds for netcdf saver (string typing)

### PR DESCRIPTION
**Pull request by marqh**
fix to make #2158 functional for newer netcdf4-python

i am not convinced this catch all potential cases, only the ones we test

a few words prior to merge would be helpful

### CODE REVIEW

**Comment by pelson** [comment]
Does ```cf_var_grid.grid_mapping_name``` need to be considered?

File: conda-requirements.txt [hunk 1]
```diff
@@ -2,9 +2,9 @@
 # conda create -n <name> --file conda-requirements.txt
 
 # Mandatory dependencies
-biggus>=0.14.0
+biggus
 cartopy
-matplotlib=1.3.1
+matplotlib
```
Comment (marqh) [inline conda-requirements.txt]: the mpl pin should not be changed in this PR

**Comment by pelson** [comment]
Python3 issue with your unit test:

```
======================================================================
FAIL: test_spherical_geog_cs (iris.tests.unit.fileformats.netcdf.test_Saver.Test__create_cf_grid_mapping)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test-environment/lib/python3.4/site-packages/Iris-1.11.0.dev0-py3.4-linux-x86_64.egg/iris/tests/unit/fileformats/netcdf/test_Saver.py", line 570, in test_spherical_geog_cs
    self._test(coord_system, expected)
  File "/home/travis/miniconda/envs/test-environment/lib/python3.4/site-packages/Iris-1.11.0.dev0-py3.4-linux-x86_64.egg/iris/tests/unit/fileformats/netcdf/test_Saver.py", line 546, in _test
    actual = self._variable_attributes(coord_system)
  File "/home/travis/miniconda/envs/test-environment/lib/python3.4/site-packages/Iris-1.11.0.dev0-py3.4-linux-x86_64.egg/iris/tests/unit/fileformats/netcdf/test_Saver.py", line 537, in _variable_attributes
    mock_grid_variable = self._grid_mapping_variable(coord_system)
  File "/home/travis/miniconda/envs/test-environment/lib/python3.4/site-packages/Iris-1.11.0.dev0-py3.4-linux-x86_64.egg/iris/tests/unit/fileformats/netcdf/test_Saver.py", line 528, in _grid_mapping_variable
    grid_variable.grid_mapping_name)
AssertionError: b'latitude_longitude' != 'latitude_longitude'
----------------------------------------------------------------------
```

**Comment by pelson** [comment]
👍

**Pull request merged**